### PR TITLE
[13.0][ADD] web_time_range_menu_custom

### PR DESCRIPTION
**Pull request by Tardo**
Adds support to set custom date ranges:

cc @tecnativa TT29249

### CODE REVIEW

File: web_time_range_menu_custom/README.rst [hunk 1]
```diff
@@ -0,0 +1,78 @@
+=============
+Web Refresher
```
Comment (chienandalu) [inline web_time_range_menu_custom/README.rst]: Not from here :slightly_smiling_face:

File: web_time_range_menu_custom/README.rst [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    :alt: Translate me on Weblate
+.. |badge5| image:: https://img.shields.io/badge/runbot-Try%20me-875A7B.png
+    :target: https://runbot.odoo-community.org/runbot/162/13.0
+    :alt: Try me on Runbot
+
+|badge1| |badge2| |badge3| |badge4| |badge5| 
+
+Adds a button next to the pager (in trees/kanban views) to refresh the displayed list.
+
+.. image:: https://raw.githubusercontent.com/OCA/web/13.0/web_refresher/static/description/refresh.png
```
Comment (chienandalu) [inline web_time_range_menu_custom/README.rst]: Delete that image as well

File: web_time_range_menu_custom/static/src/js/domain.js [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+            type,
+            comparisonPeriod,
+            periodCustom,
+            comparisonPeriodCustom
+        ) {
+            let leftBoundaryParams = {};
+            let rightBoundaryParams = {};
+            let offsetPeriodParams = 0;
+            // Cloned function from web.domain
+            // Necessary because inner function usage
```
Comment (chienandalu) [inline web_time_range_menu_custom/static/src/js/domain.js:24]: Ouch, such a rigid implementation from Odoo. It's hard to see what's monkeypatched and what's the original code. Please document in the code.

Also put some red flags and TODOs to check this in migrations to be aligned with the upstream code.

**Comment by Tardo** [comment]
Changes done

**Review by carlosdauden — APPROVED** [review]

File: web_time_range_menu_custom/README.rst [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+.. |badge4| image:: https://img.shields.io/badge/weblate-Translate%20me-F47D42.png
+    :target: https://translation.odoo-community.org/projects/web-v13/web-v13-web_time_range_menu_custom
+    :alt: Translate me on Weblate
+.. |badge5| image:: https://img.shields.io/badge/runbot-Try%20me-875A7B.png
+    :target: https://runbot.odoo-community.org/runbot/162/v13
+    :alt: Try me on Runbot
+
+|badge1| |badge2| |badge3| |badge4| |badge5| 
+
+Extend period and comparison period options for the time range menu adding a new option called "Custom Period".
```
Comment (chienandalu) [inline web_time_range_menu_custom/README.rst:28]: It'd be nice to have a brief usage section and an screenshot

**Review by chienandalu — APPROVED** [review]

**Pull request merged**